app/public/js/localSprites.js

```diff
@@ -301,66 +347,97 @@ function addFireballParticle(parent, decayFrames, parentPreScalingXSize, parentP
 function addFireballDetonation(parent, numberOfFragments, parentPreScalingXSize, parentPreScalingYSize) {
     var detonationParticles = [];
     for (var i = 0; i < numberOfFragments; i++) {
-        var newParticle = addFireballParticle(parent, 30, parentPreScalingXSize, parentPreScalingYSize, 1);
+        var newParticle = addFireballParticle(parent, 30, parentPreScalingXSize, parentPreScalingYSize, SPRITE_TYPE_FIREBALL_PARTICLE_DETONATION);
         detonationParticles.push(newParticle);
     }
     for (var j = 0; j < detonationParticles.length; j++) {
 
     }
 }
 
+//TRIGGERS
+function canTriggerTrigger(trigger) {
+    return  trigger.notReadyToTriggerUntil > now();
+}
 
-function addTrigger(left, top, width, height, target, spawnedObjectType, spawnedObjectXOffset, spawnedObjectYOffset, xForce, yForce, cooldownFrames) { //too many parameters! break function up into subtypes of trigger zone
-    //spawnedObjectXOffset and ...YOffset are where opject spawns relative to the center of the trigger zone
-    //spawnedObjectType key: 0 = doesn't spawn anything, 1 = homing fireball
+function addSpawnTrigger(left, top, width, height, cooldownInSeconds, target, spawnedObjectType, spawnedObjectXOffset, spawnedObjectYOffset) {
     var hitBox = rectangle(left, top, width, height);
     var frames = [
-        $.extend(rectangle(1 * 16, 0 * 16, 16, 16), {image: customBlocksAImage, hitBox}),
-        //$.extend(rectangle(1 * 8, 0 * 8, 8, 8), {image: fireballContrailAImage, hitBox}),
-        //$.extend(rectangle(2 * 8, 0 * 8, 8, 8), {image: fireballContrailAImage, hitBox}),
-        //$.extend(rectangle(3 * 8, 0 * 8, 8, 8), {image: fireballContrailAImage, hitBox}),
-        //$.extend(rectangle(4 * 8, 0 * 8, 8, 8), {image: fireballContrailAImage, hitBox}),
+        $.extend(rectangle(1 * 16, 0 * 16, 16, 16), {image: fireballContrailAImage, hitBox}),
+    ];
+    var spawnTrigger = new SimpleSprite({frames}, left, top, 0, 0, 1, 1);
+    spawnTrigger.type = TRIGGER_TYPE_SPAWN, //hm, don't actually use this...
+    spawnTrigger.cooldownInMS = cooldownInSeconds * 1000,
+    spawnTrigger.hitBox = hitBox,
+    spawnTrigger.notReadyToTriggerUntil = now() + spawnTrigger.cooldownInMS,
```

Comment:
Do you want it to start on cooldown? If not, you can just delete this, it will get set the first time it is triggered.